internal/controller/argocd.go

```diff
@@ -198,17 +221,17 @@ func (clctrl *ClusterController) DeployRegistryApplication() error {
 			registryPath,
 		)
 
+		if clctrl.Kcfg == nil {
+			clctrl.Kcfg = k8s.CreateKubeConfig(false, clctrl.ProviderConfig.Kubeconfig)
+		}
 
-		cmdStr := fmt.Sprintf("kubectl --kubeconfig=%s rollout restart -n argocd deploy/argocd-applicationset-controller", clctrl.ProviderConfig.Kubeconfig)
-
-		cmd := exec.Command("/bin/sh", "-c", cmdStr)
+		err = RestartDeployment(context.Background(), clctrl.Kcfg.Clientset, "argocd", "argocd-applicationset-controller")
 
-		err = cmd.Run()
 		if err != nil {
-			log.Info().Msgf("Error executing kubectl command: %v\n", err)
 			return err
 		}
 
+		log.Info().Msgf("successfull in restarting\n")
```

Comment:
Since there's no formatting directive, this will trip a handful of checks... In the same sense, these are technically a bit heavier than normal to run (whenever you call a "printf-style" function, the "stringer" interface is evaluated on every parameter which adds up).

I also don't think you need the linebreak since each log line goes in its own line, but feel free to disagree!

```suggestion
		log.Info().Msg("successfull in restarting")
```